registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -234,16 +233,23 @@ SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::computeTransfor
     }
   }
 
-  // Feature correspondence cache
+  // Precompute feature correspondence cache
   std::vector<std::vector<int> > similar_features (input_->size ());
+  std::vector<float> nn_distances(k_correspondences_);
+  #pragma omp parallel for num_threads(num_threads_) firstprivate(nn_distances)
+  for(int i = 0; i < static_cast<int>(input_->size()); ++i) {
+    nn_distances.clear();
+    feature_tree_->nearestKSearch(*input_features_, i, k_correspondences_, similar_features[i], nn_distances);
+  }
+
+  // Temporary containers
+  std::vector<int> sample_indices(nr_samples_);
+  std::vector<int> corresponding_indices(nr_samples_);
 
   // Start
+  #pragma omp parallel for num_threads(num_threads_) shared(similar_features) firstprivate(inliers, error, sample_indices, corresponding_indices)
```

Comment:
Please move error into the loop here as well as before. It's a pure out param in both places,

Reply:
Done